aws-lc-rs/src/agreement.rs

```diff
@@ -218,6 +230,14 @@ impl EphemeralPrivateKey {
                 rng.fill(&mut priv_key)?;
                 Self::from_p384_private_key(&priv_key)
             }
+            AlgorithmID::ECDH_P521 => {
+                let mut priv_key = [0u8; ECDH_P521_PRIVATE_KEY_LEN];
+                // The private key is stored in 66 bytes = 528-bits, which is too large.
```

Comment:
ok, this change makes the P-521 case the same as the others, but the general problem is still there (for all curves). For example, `priv_key` with 256 ones is not a valid P-256 private key because it's larger than the order of P-256, same for other curves.

Is this function used only for testing?

Reply:
This function is used for both production and testing.
The difference between testing and production is the type of `struct` of the `rng` passed in. In production, `rng.fill` generates a random number. In our tests, `rng.fill` is configured to pass in a known value for our KATs.
Your concern with the `priv_key` generation for all the other curves is valid though, which is why I think we should consider using `EC_KEY_generate_key` for all curves.